app/src/apdu_handler.c

```diff
@@ -213,11 +213,6 @@ __Z_INLINE void handleSign(volatile uint32_t *flags, volatile uint32_t *tx, uint
         const int error_msg_length = strnlen(error_msg, sizeof(G_io_apdu_buffer));
         MEMCPY(G_io_apdu_buffer, error_msg, error_msg_length);
         *tx += (error_msg_length);
-
-#ifdef HAVE_SWAP
-        // Finalize the transaction with failure status
-        finalize_exchange_sign_transaction(false);
-#endif 
```

Comment:
Isn't it needed anymore?

Reply:
I have added this in the last PR, but by mistake I was not running all the tests. 
This time I noticed this line is closing the communication between the exchange and app-cosmos. But it leaves the app-exchange hanging, not returning the error.